ui/components/app/permission-cell/permission-cell-status.js

```diff
@@ -111,6 +118,18 @@ export const PermissionCellStatus = ({ revoked, dateApproved, accounts }) => {
       ? t('permissionRequestedForAccounts', [renderAccountsGroup()])
       : t('permissionRequested');
 
+  let statusToShow;
+
+  if (revoked) {
+    statusToShow = statusRevoked;
+  } else if (dateApproved) {
+    statusToShow = statusApprovedOn;
+  } else if (approved) {
+    statusToShow = statusApproved;
+  } else {
+    statusToShow = statusRequestedNow;
+  }
```

Comment:
It's cleaner to extract this and the stuff above into a function and return the message.

```ts
function getStatusMessage() {
  if (revoked) {
    return accounts && accounts.length
      ? t('permissionRevokedForAccounts', [renderAccountsGroup()]);
  }

  if (dateApproved) {
    return accounts && accounts.length
      ? t('approvedOnForAccounts', [
          formatDate(dateApproved, 'yyyy-MM-dd'),
          renderAccountsGroup(),
        ])
      : t('approvedOn', [formatDate(dateApproved, 'yyyy-MM-dd')]);
  }

  // and so on
}

Reply:
That pattern is a bit hard to achieve when using a hook though 🤔

Reply:
Refactored / updated.